packages/core/src/lib/markdown-it/highlight/helper.ts

```diff
@@ -6,3 +6,61 @@ export function splitCodeAndIndentation(codeStr: string) {
   const content = codeStr.substring(codeStartIdx);
   return [indents, content];
 }
+
+export enum BOUNDARY_TYPE {
+  Start,
+  End,
+}
+
+export interface Boundary {
+  index: number;
+  type: BOUNDARY_TYPE;
+}
+
+// Simplifies multiple bounds applied on a single line to an array of disjointed bounds
```

Comment:
If this function collates overlapping intervals into one interval, can we add that to this documentation string too?

Reply:
Ah ok!